test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -76,6 +80,11 @@ public void testIncrement() {
   public void testObjLitDeclaration() {
     test("var a = {b: {}, c: {}}; var d = a.b; var e = a.c",
          "var a$b = {}; var a$c = {}; var d = a$b; var e = a$c");
+
+    test("var a = {b: {}, /** @nocollapse */ c: {}}; "
```

Comment:
If you are exporting a property and need it to remain mutable, then `@nocollapse` is needed. Restricting definitions to object literals annotated with `const` seems unnecessarily strict and would not be expected by users. 

```JavaScript
var foo = {
  /** @nocollapse */
  bar1: 10,
  bar2: 20
};
window['foo'] = window['foo'] || {};
window['foo']['bar1'] = foo.bar1;
```